### ISSUE TITLE
How to call 'apply'?

### ISSUE THREAD

**Issue by michaelaird** [nikhilk/scriptsharp#325]
I'm struggling with how to replicate a method call using "apply". I need to do something like this

```
$.when.apply($, arr);
```

but i can't figure out the correct syntax in ScriptSharp. Any thoughts?

Michael.

**Comment by nikhilk** [comment]
See the Function type. It has Apply/Call for invoking methods. You could also probably use Type.InvokeMethod/Script.InvokeMethod (depending on script# version).

**Comment by michaelaird** [comment]
I found that Function type. I just can't seem to figure out the correct C#
syntax for using it. :(

**Comment by michaelaird** [comment]
Specifically, how do I convert/cast from jQuery.When to a Function so that i can call Apply on it?

**Comment by nikhilk** [comment]
This will probably work:

```
Function f = Script.GetField<Function>(typeof(jQuery), "when");
f.Apply(...)
```

Not sure its the best though. I am still not sure why you need to use Apply to invoke it, when the API already exists as a strongly typed member on jQuery.

**Comment by michaelaird** [comment]
Apply lets you call a method with an array, that gets passed to the
function as arguments. When in particular needs needs arguments because it
will treat a single array as an object that is resolved (
http://api.jquery.com/jQuery.when/). Using Apply was the solution I found
here: http://stackoverflow.com/a/4878978.

Note that Call does not do the Array->Arguments translation, only Apply
does.

In any case, i'll try your solution tomorrow and let you know how it works.

Thanks.

**Comment by michaelaird** [comment]
Nikhil,

i'm using the published nuget package for 0.7.5 and it doesn't have Script.GetField<Function>. I think that package is  5 months old now and there have been a lot of changes. Any chance you could publish 0.7.6?

**Comment by michaelaird** [comment]
That aside, i did get it to work with 0.7.5 using this code:

```
Function f = (Function)Type.GetField(jQuery.Instance, "when");

IDeferred whenResult = (IDeferred)f.Apply(jQuery.Instance, (Object[])listOfDefferreds);

whenResult.Done(delegate(Object result)
            {
                // Do Stuff
            });
```
